### ISSUE TITLE
Loss stabilization suggestion: NaN loss + Paper questions

### ISSUE THREAD

**Issue by Magauiya** [shermanlian/spatial-entropy-loss#1]
Dear authors,

Thank you for sharing the code. I would suggest to add small constant to operations where **torch.sqrt is involved** to avoid **NaN** during gradient calculation and hence backpropagation. Also, I would appreciate if you share your thoughts on following questions:

1. Why adding Entropy loss lead to the improvement in all metrics (fidelity and perceptual)? I somehow understood the intuition behind the increase in perceptual metrics. However, it is a bit counterintuitive to me why PSNR and SSIM should improve. 
2. Do you have visual results for Tables 3 and 4? It would be great if you share those results.
3. Can you please share results for L_2 trained model vs Proposed Spatial Entropy in Figure 6? 
4. In the paper you proposed several distance calculators. Are these ablation studies and results publicly available?

Thank you!

**Comment by Algolzw** [comment]
Hi!

1. For the fidelity metrics, we believe the improvement depends on specific tasks. In the low-light enhancement case, PSNR is highly relative to the illumination similarity which can be well measured by the spatial entropy. Thus both fidelity and perceptual results were improved. However, in other tasks like image super-resolution and denoising, the results can be different.
2. We might extend this work for other image restoration tasks and will try to add more detailed tables and visual results in the future version. More distance measurements will be considered and implemented.
3.  We will also update the code as suggested!

Thank you for your great suggestions and comments!

Best.
